src/merkle_tree.ml

```diff
@@ -456,6 +456,32 @@ struct
     in
     implied_root next_entry_hash addr0 prev_path
 
+  let%snarkydef_ modify_or_get_req ~(depth : int) root addr0 ~f :
```

Comment:
This and `modiify_req` look to be nearly identical.. can we combine them and just have `modify_req` discard the `prev` part of the return value? They also both seem to include the entire definition of `get_req`, which should probably be a function call instead.

Reply:
Sure, I agree those two functions look really similar. And `modify_or_get_req` besides doing the modification, also returns the old value. I delete the origin `modify_req` and replace it with the new one. I did it on a different branch though. https://github.com/o1-labs/snarky/pull/246. Please take a look at that one.